app/controllers/drivers_controller.rb

```diff
@@ -0,0 +1,95 @@
+class DriversController < ApplicationController
+
+  def index
+    @drivers = Driver.all
+  end
+
+  def show
+    @driver = Driver.find_by(id: params[:id])
+    if @driver.nil?
+      head :not_found
+      return
+    end
+  end
+
+  def new
+    @driver = Driver.new
+  end
+
+  # this method makes the driver default to available (true)
+  def driver_created
+    new_driver = driver_params
+    if new_driver[:available].nil?
+      new_driver[:available] = true
+    end
+
+    return new_driver
+  end
+
+  def create
+    @driver = Driver.new(driver_created) 
+    if @driver.save 
+      flash[:success] = "#{@driver.name.capitalize} was successfully added!"
+      redirect_to driver_path(@driver.id) 
+      return
+    else
+      flash.now[:error] = "#{@driver.name.capitalize} was NOT successfully added!"
+      render :new, status: :bad_request
+      return
```

Comment:
This error causes problems if the driver doesn't exist! consider changing to this:
```suggestion
      if @ driver != nil
          flash.now[:error] = "#{@driver.name.capitalize} was NOT successfully added!"
      else
          flash.now[:error] = "Something went wrong: Invalid Driver"
      end
      
      render :new, status: :bad_request
      return
```